### PR TITLE
fix: egs unable to login (fixes #4472)

### LINKED ISSUE #4472: Can't use Epic Games Store as source
Lutris no longer lists games from the Epic Games Store.
In fact, after disconnecting, further attempts to reconnect to an Epic Games account from Lutris don't seem to do anything: after logging in, the authentication window closes and nothing else happens.

Seen on:
- Lutris version: 0.5.10.1
- OS,DE: Pop!_OS 22.04, Gnome 42.3.1

Edit: [Somebody else has also reported this issue on the forums](https://forums.lutris.net/t/cant-connect-epic-games-as-source/16118).

### PR DESCRIPTION
**Pull request by Relms12345**
Fixes #4472

### CODE REVIEW

**Comment by Joeao** [comment]
When running this I faced the same issue as explained in #4472.
@Relms12345 Can you confirm that running this change gave you positive results?

**Comment by Relms12345** [comment]
> When running this I faced the same issue as explained in #4472.
> @Relms12345 Can you confirm that running this change gave you positive results?

Yea when running on both a clean and existing install of lutris it gave me positive results and I was able to download and run HITMAN3

File: lutris/services/egs.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ class EpicGamesStoreService(OnlineService):
     cookies_path = os.path.join(settings.CACHE_DIR, ".egs.auth")
     token_path = os.path.join(settings.CACHE_DIR, ".egs.token")
     cache_path = os.path.join(settings.CACHE_DIR, "egs-library.json")
-    login_url = "https://www.epicgames.com/id/login?redirectUrl=https://www.epicgames.com/id/api/redirect"
+    login_url = "https://legendary.gl/epiclogin"
```
Comment (gucci-on-fleek) [inline lutris/services/egs.py:153]: Is there a way to do this _without_ relying on a third-party server?

Comment (Relms12345) [reply lutris/services/egs.py]: yep fixed

**Comment by gucci-on-fleek** [comment]
I've tested this and can confirm that it works. (Fedora 36, Python 3.10.6)

**Comment by danieljohnson2** [comment]
I'd merge this one, but it looks like it's still in flux.

I'll give it more time to settle down, but I think we want this for the next release.

**Comment by JustAFriendlyCat** [comment]
This PR fixed the issue for me(Linux Mint 21, Python 3.10.4)

**Comment by Inve1951** [comment]
fix is confirmed on my end
`Linux 5.16.0-3mx-amd64 #1 SMP PREEMPT Debian 5.16.11-1~mx21+1`
`Python 3.9.2`

**Comment by danieljohnson2** [comment]
Well, I wish @Relms12345 was talking to me, but I guess he's off saving the free world or something like that. Kids these days.

I think we need this fix, so I'll go ahead and merge it anyway.

**Comment by gucci-on-fleek** [comment]
@xMukkhx 

### Option 1 (Recommended)

1. Install Git. Most distros have this included, but if it isn't, you should install with your package manager:
    ```sh
    sudo apt install git # Ubuntu, Debian, etc.
    sudo dnf install git # Fedora, CentOS, etc.
    sudo pacman -Syu git # Arch, Manjaro, etc.
    ```
2. Clone lutris:
    ```sh
    cd ~
    git clone --depth 1 https://github.com/lutris/lutris.git
3. Run lutris
    ```sh
    cd ~/lutris
    ./bin/lutris
    ```
    (Run this step every time that you launch Lutris)

### Option 2 (Not recommended)

1. Install Git. Most distros have this included, but if it isn't, you should install with your package manager:
    ```sh
    sudo apt install git # Ubuntu, Debian, etc.
    sudo dnf install git # Fedora, CentOS, etc.
    sudo pacman -Syu git # Arch, Manjaro, etc.
    ```
2. Run the following commands:
    ```sh
    cd /usr/lib/python3*/site-packages
    curl https://patch-diff.githubusercontent.com/raw/lutris/lutris/pull/4492.diff | sudo git apply
    ```
3. Launch Lutris as normal

**Comment by wolfmanx** [comment]
For Ubuntu it is:

>     2. Run the following commands:
>        ```shell
>        cd /usr/lib/python3*/dist-packages
>        curl https://patch-diff.githubusercontent.com/raw/lutris/lutris/pull/4492.diff | sudo git apply
>        ```

**Comment by tairosonloa** [comment]
When will this fix will released? I'm having the same issue this should fix in Lutris 0.5.11 (flatpak in a Steam Deck)

**Comment by danieljohnson2** [comment]
Soon, I hope; we've got several issues like this fixed in master now.

But it is up to @strycore.

**Pull request merged**